### ISSUE TITLE
Block Ordering Bug $this->orderBy('neoblocks.sortOrder') not being set.

### ISSUE THREAD

**Issue by philfuse** [spicywebau/craft-neo#463]
#### Description
When ordering blocks we're seeing an issue that in the preview the blocks are not showing in the correct order.
We believe this is due to a check in the code around setting the sort order
elements/db/BlockQuery.php line 370 
We think it is assuming a sequential array but it is in fact an associative array and so the sort order is never set.

#### Steps to reproduce
1. change the order or elements  https://www.dropbox.com/s/zm3v1hmjs177kt5/Screenshot%202021-06-07%20at%2013.55.52.png?dl=0
2. $this-select is not sequential array https://www.dropbox.com/s/sacag27mteesnux/Screenshot%202021-06-07%20at%2013.59.36.png?dl=0
3. the check fails https://www.dropbox.com/s/xywb71kefqhbqhw/Screenshot%202021-06-07%20at%2014.00.13.png?dl=0
4. the resultant orderBy is never set.

A potential complex fix would be to check the array type i think?
https://www.dropbox.com/s/1e4ra6shwe246ic/Screenshot%202021-06-07%20at%2014.01.53.png?dl=0

Or a much simpler fix would be to just always assume if blockHasSortOrder is true always set the sorting 
https://www.dropbox.com/s/92e43bafgcx74sr/Screenshot%202021-06-07%20at%2014.02.33.png?dl=0

The code looks like it is just trying to avoid doing a sort if a COUNT is being requested which makes sense but perhaps doesn't hurt to always add the sort.

#### Other information
- Neo version: 2.9.5
- Craft version: 3.6.11.2
- (if multi-site) What is the affected Neo field's propagation method:
- Is eager-loading used?

**Comment by ttempleton** [comment]
The preview block ordering seems to be working fine on my end, but that is a clear bug with the `$this->select` check in any case.

In my testing it does seem like `COUNT(*)` is only included if it is a sequential array, though, so we should be safe to change

```php
isset($this->select[0]) && $this->select[0] !== 'COUNT(*)'
```

to

```php
(!isset($this->select[0]) || $this->select[0] !== 'COUNT(*)') && $this->structureId !== null
```

(also adding the `$this->structureId` check because it seems unnecessary to order by `sortOrder` if the blocks don't belong to the same structure)

Does that change resolve the block ordering issue for you?

**Comment by philfuse** [comment]
hi @ttempleton 
thanks for quick response.... !
this does indeed look to fix the issue - smart!
We can run that patch in locally during dev.
Will this go into next release you think?

**Comment by ttempleton** [comment]
Awesome, thanks for confirming that.  Just released Neo 2.9.12 with that fix included.
